[EDU-4205] feat: add ref on s3 protocol and guide on s3cmd (#887)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/store/object-storage/index.mdx

| Class | Operation name | HTTP method | S3 capability |
| --- | --- | --- | --- |
| A | [ListObjects](#listobjects) | `GET` | `listFiles`<br />`listBuckets`* |
| A | [CreateBucket](#createbucket) | `POST` | – |
| A | [ListBuckets](#listbuckets) | `GET` | `listAllBucketNames` |
| A | [UpdateBucket](#updatebucket) | `PATCH` | – |
| B | [GetObject](#getobject) | `GET` | `readFiles` |
| C | [PostObject](#postobject) | `POST` | – |
| C | [PutObject](#putobject) | `PUT` | `writeFiles` |
| C | [DeleteObject](#deleteobject) | `DELETE` | `deleteFiles` |
| C | [DeleteBucket](#deletebucket) | `DELETE` | – |

> \* If `listBuckets` is enabled, when attempting to retrieve files that aren't in the bucket using an S3 credential, the proper `404 Not Found` status response returns instead of a `403 Forbidden` status. Find out more about S3 capabilities in [S3 protocol compatibility](#s3-protocol-compatibility).

### PostObject

Uploads an object to a bucket.

:::tip
For API cURL requests, use the `--data-binary` option in the command to send raw binary data, such as images or complex files without undergoing URL encoding.
:::

> For the **Azion API**, you can specify the MIME type of the object being sent in the body using the `Content-Type` header. For example, objects with the `.txt` extension should contain the `Content-Type: text/plain` header. If the MIME type is not specified, **Edge Storage** will attempt to interpret the file type based on the file extension. Alternatively, use the `application/octet-stream` MIME type to indicate that the data is a binary stream and the server should handle it as raw binary data.

Reviewer: ```suggestion
For the **Azion API**, you can specify the MIME type of the object being sent in the body using the `Content-Type` header. For example, objects with the `.txt` extension should contain the `Content-Type: text/plain` header. If the MIME type isn't specified, **Edge Storage** will attempt to interpret the file type based on the file extension. Alternatively, use the `application/octet-stream` MIME type to indicate that the data is a binary stream and the server should handle it as raw binary data.
```

:::

### Bucket permissions

In addition to the required authentication and necessary permissions, some API operations can be restricted by bucket permissions. The permissions available are:

- **Read-only**: objects in the bucket can be read but not modified.
- **Read-write**: objects in the bucket can be modified.
- **Restricted**: objects in the bucket can be modified and read but can't be accessed by the Azion Edge Platform. Buckets can't be modified using Edge Runtime and can't be used as an origin for edge applications.

### S3 credentials

**Edge Storage** offers compatibility with the S3 protocol through credentials. 

Credentials can be created for any bucket that you own or for your account as a whole. With them, you can control permissions for operations associated with that credential. The permissions for the credential are exclusive to access thorugh the S3 protocol.

> Once a credential is created, an *access key* and a *secret key* are generated, which can be used to set up access to the bucket through the S3 protoco. For security reasons, the *secret key* won't be available after the credential is created. Existing credentials cannot be modified in any way.

Reviewer: ```suggestion
Once a credential is created, an *access key* and a *secret key* are generated, which can be used to set up access to the bucket through the S3 protoco. For security reasons, the *secret key* won't be available after the credential is created. Existing credentials can't be modified in any way.
```

> Once a user's access is verified, they are allowed to make operations depending on the capabilities set for the credential. 

Reviewer: ```suggestion
Once a user's access is verified, they're allowed to make operations depending on the capabilities set for the credential. 
```

- `listAllBucketNames`: equivalent to [ListBuckets](#listbuckets), allows you to list all buckets associated with the account.
- `listBuckets`: if enabled, returns the proper `404 Not Found` response when attempting to retrieve files that aren't in the bucket using the credential.

---

## S3 protocol compatibility

After an [S3 credential](#s3-credentials) is created for a bucket, you can use the S3 protocol (`s3://`) to execute operations according to the [list of capabilities](#capabilities). 

The S3 protocol allows you to access buckets and objects using an Edge Storage URL. This configuration facilitates file operations through command line interface (CLI) tools, such as [s3cmd](https://s3cmd.org), database services, or functions.

You can use the access and secret keys provided by the [S3 credentials API](https://api.azion.com) to set up a connection using the S3 protocol.

<Button href="/en/documentation/products/guides/s3-protocol-for-edge-storage/" text="learn How to access Edge Storage using the S3 protocol" variant="secondary"/>

To do so, you'll need the following information:

| Data | Description |
| --- | --- |

> | Access key | The credential's access key generated upon creating the S3 credential with the Azion API. |
> | Secret key | The credential's secret key generated upon creating the S3 credential with the Azion API. This information is confidential and will only be available at the moment of creation. |
> | Region | The assigned server's region, which is `us-east-005`. |
> | S3 endpoint | The default S3 address for all operations, which is `s3.us-east-005.azionstorage.net`. |
> | DNS-style template | The host name template to access the bucket and objects. Can be `bucket+hostname:port/object-key` or `hostname:port/bucket`. <br /><br />For example, for a `file.txt` object in the `my-bucket` bucket, the host names could be: <br /><li><ul>`my-bucket.s3.us-east-005.azionstorage.net/file.txt`</ul><ul>`s3.us-east-005.azionstorage.net/my-bucket/file.txt`</ul></li> |

Reviewer 2: ```suggestion
| Access key | The credential's access key generated upon creating the S3 credential with the Azion API |
| Secret key | The credential's secret key generated upon creating the S3 credential with the Azion API. This information is confidential and will only be available at the moment of creation |
| Region | The assigned server's region, which is `us-east-005` |
| S3 endpoint | The default S3 address for all operations, which is `s3.us-east-005.azionstorage.net` |
| DNS-style template | The host name template to access the bucket and objects. Can be `bucket+hostname:port/object-key` or `hostname:port/bucket`. <br /><br />For example, for a `file.txt` object in the `my-bucket` bucket, the host names could be: <br /><li><ul>`my-bucket.s3.us-east-005.azionstorage.net/file.txt`</ul><ul>`s3.us-east-005.azionstorage.net/my-bucket/file.txt`</ul></li> |
```
